presto-spark-base/src/main/java/com/facebook/presto/spark/execution/HttpNativeExecutionTaskResultFetcher.java

```diff
@@ -119,20 +121,28 @@ public CompletableFuture<Void> start()
                 });
     }
 
+    public void stop()
+    {
+        if (schedulerFuture != null) {
+            schedulerFuture.cancel(false);
+        }
+    }
+
     /**
      * Blocking call to poll from result buffer. Blocks until content becomes
      * available in the buffer, or until timeout is hit.
      *
      * @return the first SerializedPage result buffer contains.
      */
-    public SerializedPage pollPage()
+    public Optional<SerializedPage> pollPage()
             throws InterruptedException
     {
         SerializedPage page = pageBuffer.poll((long) POLL_TIMEOUT.getValue(), POLL_TIMEOUT.getUnit());
         if (page != null) {
             bufferMemoryBytes.addAndGet(-page.getSizeInBytes());
```

Comment:
Do we have unit test to assert `bufferMemoryBytes` will return to 0 when all pages are polled?

Reply:
That is in the plan, and will come in the following commits